### PR TITLE
[oneDPL] Fix include guard names in header files

### PR DESCRIPTION
**Pull request by SergeyKopienko**
Normalized include guard names in header files:
- in file https://github.com/oneapi-src/oneDPL/blob/dev/skopienko/fix_include_guard_names_in_oneapi_dpl_headers/.clang-format we have next line: **BasedOnStyle: LLVM**
- This line mean that we should create header file guards as described in 
[Header Guard](https://llvm.org/docs/CodingStandards.html#id10)(https://llvm.org/docs/CodingStandards.html#header-guard)
_The header file’s guard should be the all-caps path that a user of this header would #include, using ‘_’ instead of path separator and extension marker. For example, the header file llvm/include/llvm/Analysis/Utils/Local.h would be #include-ed as #include "llvm/Analysis/Utils/Local.h", so its guard is LLVM_ANALYSIS_UTILS_LOCAL_H._

### CODE REVIEW

File: include/oneapi/dpl/algorithm [hunk 1]
```diff
@@ -13,8 +13,8 @@
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef _ONEDPL_ALGORITHM
```
Comment (MikeDvorskiy) [inline include/oneapi/dpl/algorithm:16]: I have a concern that variable can be used in the some code for some workaround somewhere....
(for example To check OnDPL was included or not..)

Comment (SergeyKopienko) [inline include/oneapi/dpl/algorithm:16]: At all gitHUB this define is used only our code.

File: include/oneapi/dpl/async [hunk 2]
```diff
@@ -13,8 +13,8 @@
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef _ONEDPL_ASYNC
```
Comment (SergeyKopienko) [inline include/oneapi/dpl/async:16]: At all gitHUB this define used only our code.

File: include/oneapi/dpl/execution [hunk 3]
```diff
@@ -13,8 +13,8 @@
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef _ONEDPL_EXECUTION
```
Comment (SergeyKopienko) [inline include/oneapi/dpl/execution:16]: At all gitHUB this define is used only our code.

File: include/oneapi/dpl/iterator [hunk 4]
```diff
@@ -13,8 +13,8 @@
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef _ONEDPL_ITERATOR
```
Comment (SergeyKopienko) [inline include/oneapi/dpl/iterator:16]: * At all gitHUB this define is used only our code.

File: include/oneapi/dpl/memory [hunk 5]
```diff
@@ -13,8 +13,8 @@
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef _ONEDPL_MEMORY
```
Comment (SergeyKopienko) [inline include/oneapi/dpl/memory:16]: * At all gitHUB this define is used only our code.

File: include/oneapi/dpl/numeric [hunk 6]
```diff
@@ -13,8 +13,8 @@
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef _ONEDPL_NUMERIC
```
Comment (SergeyKopienko) [inline include/oneapi/dpl/numeric:16]: * At all gitHUB this define is used only our code.

File: include/oneapi/dpl/algorithm (hunk 1 above)
Comment (SergeyKopienko) [reply include/oneapi/dpl/algorithm:16]: I this that your concern is too strict: I am tried too find all of changed defined at all gitHub and they was found only in our code.

**Comment by SergeyKopienko** [comment]
@rarutyun please make review too

**Comment by SergeyKopienko** [comment]
@rarutyun please make review too

**Comment by SergeyKopienko** [comment]
@rarutyun could you please take a look at this PR ?

**Pull request closed without merging**